### PR TITLE
Migrate Server cert manager UpstreamAuthority to V1

### PR DESCRIPTION
**Pull request by MarcosDY**

### CODE REVIEW

File: pkg/server/plugin/upstreamauthority/certmanager/certmanager.go [hunk 1]
```diff
@@ -161,25 +189,25 @@ func (p *Plugin) MintX509CA(request *upstreamauthorityv0.MintX509CARequest, stre
...
 			log.Error("Failed to wait for CertificateRequest to become ready in time")
-			return errors.New("request did not become ready in time")
+			return status.Error(codes.Internal, "request did not become ready in time")
 		}
 
 		time.Sleep(time.Second / 4)
 
 		if err := p.cmclient.Get(ctx, obj, cr); err != nil {
-			return err
+			return status.Errorf(codes.Internal, "kubernet cluster client filed to get: %v", err)
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/certmanager/certmanager.go:198]:
```suggestion
			return status.Errorf(codes.Internal, "kubernetes cluster client failed to get object: %v", err)
```

File: pkg/server/plugin/upstreamauthority/certmanager/certmanager.go [hunk 2]
```diff
@@ -245,24 +275,7 @@ func getKubeConfig(configPath string) (*rest.Config, error) {
...
-	return makeError(codes.Unimplemented, "publishing upstream is unsupported")
-}
-
-func makeError(code codes.Code, format string, args ...interface{}) error {
-	return status.Errorf(code, "cert-manager: "+format, args...)
-}
-
-func (*Plugin) GetPluginInfo(context.Context, *spi.GetPluginInfoRequest) (*spi.GetPluginInfoResponse, error) {
-	return &spi.GetPluginInfoResponse{}, nil
+func (*Plugin) PublishJWTKeyAndSubscribe(*upstreamauthorityv1.PublishJWTKeyRequest, upstreamauthorityv1.UpstreamAuthority_PublishJWTKeyAndSubscribeServer) error {
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/certmanager/certmanager.go:279]: Do you mind grouping this close to `MintX509CAAndSubscribe`?

File: pkg/server/plugin/upstreamauthority/certmanager/certmanager_test.go [hunk 3]
```diff
@@ -51,112 +54,88 @@ func Test_MintX509CA(t *testing.T) {
...
 	tests := map[string]struct {
-		request              *upstreamauthorityv0.MintX509CARequest
+		csr                  []byte
+		preferredTTL         time.Duration
 		updateCR             func(t *testing.T, cr *cmapi.CertificateRequest)
-		expX509CaChain       [][]byte
-		expUpstreamX509Roots [][]byte
-		expError             bool
+		expX509Ca            []*x509.Certificate
+		expUpstreamX509Roots []*x509.Certificate
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/certmanager/certmanager_test.go:61]: can we name these consistent with `expectCode` and `expectMsgPrefix`? i.e. expectX509CA and expectUpstreamX509Roots

File: pkg/server/plugin/upstreamauthority/certmanager/certmanager_test.go [hunk 4]
```diff
@@ -214,87 +202,178 @@ func Test_MintX509CA(t *testing.T) {
...
+			newClient: func(configPath string) (client.Client, error) {
+				return cmclient, nil
+			},
+			onCreateCR: func() {
+				crCreated <- struct{}{}
+			},
+			onCleanupStaleCRs: func() {
+				staleCRsDeleted <- struct{}{}
+			},
+		},
```
Comment (azdagron) [inline pkg/server/plugin/upstreamauthority/certmanager/certmanager_test.go:356]: do we need all these hooks to be set up for testing out the JWT key stuff?

**Review by azdagron — COMMENTED** [review]
Thank you for the quick turn around on this @MarcosDY ! Just a few comments...

File: pkg/server/plugin/upstreamauthority/certmanager/certmanager_test.go (hunk 4 above)
Comment (MarcosDY) [reply pkg/server/plugin/upstreamauthority/certmanager/certmanager_test.go:356]: no, only cmclient. I'll remove anothers

**Comment by azdagron** [comment]
I think we're ready to take this once the merge conflict has been resolved. Thanks, @MarcosDY !

**Review by azdagron — APPROVED** [review]

**Pull request merged**